diff --git a/bl2/aarch32/bl2_el3_entrypoint.S b/bl2/aarch32/bl2_el3_entrypoint.S
index 9b4da6b13..044200475 100644
--- a/bl2/aarch32/bl2_el3_entrypoint.S
+++ b/bl2/aarch32/bl2_el3_entrypoint.S
@@ -89,3 +89,4 @@ func bl2_run_next_image
 	ldm	r8, {r0, r1, r2, r3}
 	eret
 endfunc bl2_run_next_image
+.section .note.GNU-stack,"",%progbits
diff --git a/bl2/aarch32/bl2_el3_exceptions.S b/bl2/aarch32/bl2_el3_exceptions.S
index dff4e36a4..493582f64 100644
--- a/bl2/aarch32/bl2_el3_exceptions.S
+++ b/bl2/aarch32/bl2_el3_exceptions.S
@@ -28,3 +28,4 @@ vector_base bl2_vector_table
 	b	report_exception	/* Reserved */
 	b	report_exception	/* IRQ */
 	b	report_exception	/* FIQ */
+.section .note.GNU-stack,"",%progbits
diff --git a/bl32/sp_min/aarch32/entrypoint.S b/bl32/sp_min/aarch32/entrypoint.S
index c4a6e8e83..b8ce2f6f9 100644
--- a/bl32/sp_min/aarch32/entrypoint.S
+++ b/bl32/sp_min/aarch32/entrypoint.S
@@ -316,3 +316,4 @@ endfunc sp_min_warm_entrypoint
 func sp_min_exit
 	monitor_exit
 endfunc sp_min_exit
+.section .note.GNU-stack,"",%progbits
diff --git a/common/aarch32/debug.S b/common/aarch32/debug.S
index a058645e3..54adf43b4 100644
--- a/common/aarch32/debug.S
+++ b/common/aarch32/debug.S
@@ -232,3 +232,4 @@ func asm_print_hex
 	bne	1b
 	bx	r7
 endfunc asm_print_hex
+.section .note.GNU-stack,"",%progbits
diff --git a/drivers/st/uart/aarch32/stm32_console.S b/drivers/st/uart/aarch32/stm32_console.S
index 0fabcc82e..8e6dc0e33 100644
--- a/drivers/st/uart/aarch32/stm32_console.S
+++ b/drivers/st/uart/aarch32/stm32_console.S
@@ -242,3 +242,4 @@ func console_stm32_flush
 	ldr	r0, [r0, #CONSOLE_T_STM32_BASE]
 	b	console_stm32_core_flush
 endfunc console_stm32_flush
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/aarch32/cache_helpers.S b/lib/aarch32/cache_helpers.S
index 7cbefe6ea..bd83b7cff 100644
--- a/lib/aarch32/cache_helpers.S
+++ b/lib/aarch32/cache_helpers.S
@@ -217,3 +217,4 @@ endfunc dcsw_op_level2
 func dcsw_op_level3
 	dcsw_op_level #(3 << LEVEL_SHIFT)
 endfunc dcsw_op_level3
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/aarch32/misc_helpers.S b/lib/aarch32/misc_helpers.S
index 6d2ec1c52..77ab7bde7 100644
--- a/lib/aarch32/misc_helpers.S
+++ b/lib/aarch32/misc_helpers.S
@@ -188,3 +188,4 @@ func disable_mmu_icache_secure
 	ldr	r1, =(SCTLR_M_BIT | SCTLR_C_BIT | SCTLR_I_BIT)
 	b	do_disable_mmu
 endfunc disable_mmu_icache_secure
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/compiler-rt/builtins/arm/aeabi_ldivmod.S b/lib/compiler-rt/builtins/arm/aeabi_ldivmod.S
index 038ae5d72..aca2625ac 100644
--- a/lib/compiler-rt/builtins/arm/aeabi_ldivmod.S
+++ b/lib/compiler-rt/builtins/arm/aeabi_ldivmod.S
@@ -43,4 +43,4 @@ DEFINE_COMPILERRT_FUNCTION(__aeabi_ldivmod)
 END_COMPILERRT_FUNCTION(__aeabi_ldivmod)
 
 NO_EXEC_STACK_DIRECTIVE
-
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/compiler-rt/builtins/arm/aeabi_uldivmod.S b/lib/compiler-rt/builtins/arm/aeabi_uldivmod.S
index be343b6bc..a6403b3fb 100644
--- a/lib/compiler-rt/builtins/arm/aeabi_uldivmod.S
+++ b/lib/compiler-rt/builtins/arm/aeabi_uldivmod.S
@@ -43,4 +43,4 @@ DEFINE_COMPILERRT_FUNCTION(__aeabi_uldivmod)
 END_COMPILERRT_FUNCTION(__aeabi_uldivmod)
 
 NO_EXEC_STACK_DIRECTIVE
-
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/cpus/aarch32/cortex_a7.S b/lib/cpus/aarch32/cortex_a7.S
index 4d4bb77bb..5ccf5ae49 100644
--- a/lib/cpus/aarch32/cortex_a7.S
+++ b/lib/cpus/aarch32/cortex_a7.S
@@ -82,3 +82,4 @@ declare_cpu_ops cortex_a7, CORTEX_A7_MIDR, \
 	cortex_a7_reset_func, \
 	cortex_a7_core_pwr_dwn, \
 	cortex_a7_cluster_pwr_dwn
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/cpus/aarch32/cpu_helpers.S b/lib/cpus/aarch32/cpu_helpers.S
index 9b5d787ed..9e7553586 100644
--- a/lib/cpus/aarch32/cpu_helpers.S
+++ b/lib/cpus/aarch32/cpu_helpers.S
@@ -246,3 +246,4 @@ func print_errata_status
 	pop	{r4, r5, r12, pc}
 endfunc print_errata_status
 #endif
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/el3_runtime/aarch32/cpu_data.S b/lib/el3_runtime/aarch32/cpu_data.S
index bdad2c145..bcfece8e8 100644
--- a/lib/el3_runtime/aarch32/cpu_data.S
+++ b/lib/el3_runtime/aarch32/cpu_data.S
@@ -40,3 +40,4 @@ func _cpu_data_by_index
 	add	r0, r0, r1
 	bx	lr
 endfunc _cpu_data_by_index
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/locks/exclusive/aarch32/spinlock.S b/lib/locks/exclusive/aarch32/spinlock.S
index 9492cc081..d80d8773e 100644
--- a/lib/locks/exclusive/aarch32/spinlock.S
+++ b/lib/locks/exclusive/aarch32/spinlock.S
@@ -41,3 +41,4 @@ func spin_unlock
 	COND_SEV()
 	bx	lr
 endfunc spin_unlock
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/psci/aarch32/psci_helpers.S b/lib/psci/aarch32/psci_helpers.S
index 5cc192e66..32dd45374 100644
--- a/lib/psci/aarch32/psci_helpers.S
+++ b/lib/psci/aarch32/psci_helpers.S
@@ -146,3 +146,4 @@ func psci_power_down_wfi
 	wfi
 	no_ret	plat_panic_handler
 endfunc psci_power_down_wfi
+.section .note.GNU-stack,"",%progbits
diff --git a/lib/xlat_tables_v2/aarch32/enable_mmu.S b/lib/xlat_tables_v2/aarch32/enable_mmu.S
index f2fff36c2..32764d822 100644
--- a/lib/xlat_tables_v2/aarch32/enable_mmu.S
+++ b/lib/xlat_tables_v2/aarch32/enable_mmu.S
@@ -118,3 +118,4 @@ func enable_mmu_direct_hyp
 
 	bx	lr
 endfunc enable_mmu_direct_hyp
+.section .note.GNU-stack,"",%progbits
diff --git a/plat/common/aarch32/platform_helpers.S b/plat/common/aarch32/platform_helpers.S
index d2c20b4a7..fcd0b2e08 100644
--- a/plat/common/aarch32/platform_helpers.S
+++ b/plat/common/aarch32/platform_helpers.S
@@ -101,3 +101,4 @@ endfunc bl1_plat_prepare_exit
 func plat_panic_handler
 	b	plat_panic_handler
 endfunc plat_panic_handler
+.section .note.GNU-stack,"",%progbits
diff --git a/plat/common/aarch32/platform_mp_stack.S b/plat/common/aarch32/platform_mp_stack.S
index 6c3d08de3..d61efdf0a 100644
--- a/plat/common/aarch32/platform_mp_stack.S
+++ b/plat/common/aarch32/platform_mp_stack.S
@@ -45,3 +45,4 @@ endfunc plat_set_my_stack
 	 */
 declare_stack platform_normal_stacks, tzfw_normal_stacks, \
 		PLATFORM_STACK_SIZE, PLATFORM_CORE_COUNT
+.section .note.GNU-stack,"",%progbits
diff --git a/plat/common/aarch32/platform_up_stack.S b/plat/common/aarch32/platform_up_stack.S
index 836c13ae6..364e96ac5 100644
--- a/plat/common/aarch32/platform_up_stack.S
+++ b/plat/common/aarch32/platform_up_stack.S
@@ -45,3 +45,4 @@ endfunc plat_set_my_stack
 	 */
 declare_stack platform_normal_stacks, tzfw_normal_stacks, \
 		PLATFORM_STACK_SIZE, 1, CACHE_WRITEBACK_GRANULE
+.section .note.GNU-stack,"",%progbits
diff --git a/plat/st/stm32mp1/stm32mp1_helper.S b/plat/st/stm32mp1/stm32mp1_helper.S
index b80716253..251ed1322 100644
--- a/plat/st/stm32mp1/stm32mp1_helper.S
+++ b/plat/st/stm32mp1/stm32mp1_helper.S
@@ -382,3 +382,4 @@ func wfi_svc_int_enable
 	pop	{r4,r8,lr}
 	bx	lr
 endfunc wfi_svc_int_enable
+.section .note.GNU-stack,"",%progbits
